src/test/java/unit/FineGrainedAccessGcsObjectReference.java

```diff
@@ -412,37 +444,32 @@ void addRefWithPartialAccess() throws IOException {
         "--bucket-name=" + externalBucket.getName());
 
     // check that the object is in the list
-    List<UFGcsObject> matchedResourceList = listObjectResourceWithName(name);
+    List<UFGcsObject> matchedResourceList = listObjectResourceWithName(withAccessBlobName);
     assertEquals(1, matchedResourceList.size());
+
+    // `terra resource delete --name=$name`
+    TestCommand.runCommandExpectSuccess(
+        "resource", "delete", "--name=" + withAccessBlobName, "--quiet");
+
+    cleanUp(Optional.of("WRITER"), Optional.empty());
   }
 
-  @Test
-  @DisplayName("describe reference to a bucket object that the user has READER access to")
-  void describeObjectReferenceWithNoAccess() throws IOException {
+  /**
+   * Reset the role of the shareeUser and remove the resource if the shareeUser is not able to
+   * remove it.
+   */
+  private void cleanUp(Optional<String> role, Optional<String> resourceName) throws IOException {
```

Comment:
There's no guarantee this method will run in each test method. e.g. if an assertion fails between the `add-user` and `cleanUp` calls. Since you are using the same `shareeUser` in all test methods, you can just grant them the role you want in `setupOnce`.

If you need to test different roles for the `shareeUser`, then I would make a new workspace for each test method that needs this, so that you can be sure the tests won't conflict with each other.

And regarding the resource deletion, it's ok if that doesn't happen for each test because the resources will get cleaned up when the workspace is deleted in `cleanupOnce`. You do need to make sure that each test method uses distinct resource name(s) to avoid collision though.

Reply:
done.